k8s/monitoring/kubemon.py

```diff
@@ -0,0 +1,100 @@
+"""
+Monitor Global.health Kubernetes pods to see if any
+are stuck in Pending status for a while. If such pods are
+present, notify on Slack.
+
+IMPORTANT:
+  The kubernetes Python package needs to be kept in sync
+  with the EKS Kubernetes version. The compatibility matrix
+  for the client Python library is at
+  https://github.com/kubernetes-client/python#compatibility
+
+NOTE:
+  This will need to be updated once we move to separate
+  clusters for different environments.
+"""
+
+import os
+import sys
+import base64
+import logging
+import datetime
+from typing import Any
+from pathlib import Path
+
+import requests
+import kubernetes
+
+def setup_logger():
+    h = logging.StreamHandler(sys.stdout)
+    rootLogger = logging.getLogger()
+    rootLogger.addHandler(h)
+    rootLogger.setLevel(logging.INFO)
+
+
+SERVICES = ["curator", "data", "location"]
+SLACK_WEBHOOK_URL = os.getenv("SLACK_WEBHOOK_URL")
+
+if (
+    kubeconfig_path := Path.home() / ".kube" / "config"
+).exists() and kubeconfig_path.stat().st_size > 0:
+    logging.info("Using existing ~/.kube/config")
+elif KUBECONFIG_VALUE := os.getenv("KUBECONFIG_VALUE"):
+    logging.info("Writing base64 encoded KUBECONFIG_VALUE to ~/.kube/config")
+    (Path.home() / ".kube").mkdir(exist_ok=True)
+    logging.info(kubeconfig_path)
+    kubeconfig_path.write_text(base64.b64decode(KUBECONFIG_VALUE).decode("utf-8"))
+elif os.getenv("ENV") == "test":
+    logging.info("Skipping kubernetes configuration in testing")
+else:
+    logging.error("Neither ~/.kube/config or KUBECONFIG_VALUE exists, aborting")
+    sys.exit(1)
```

Comment:
Put this in a function, otherwise if someone imports this module to use a different function they will overwrite their Kubernetes config.